src/App.vue

```diff
@@ -338,6 +338,7 @@ export default {
 		}
 
 		register(await connect())
+		emit('talk:audioEncoder:ready')
```

Comment:
wouldn't it be possible to check and wait for encoder state only in NewMessageForm or deeper instead of globally ?

Reply:
The only 2 solutions that came up to me are a store or an event. I wouldn´t know how else to do it. And store seemed overkill. Do you have any ideas?

Reply:
can't you move register/connect to the NewMessageForm's in its mounted function ?

Reply:
and if it can be registered only once, store the fact that it was already registered.

one way would be also to wrap it in a service JS file that only initializes once ever.
you could all its initialize function then in NewMessageForm's mounted or even closer to the place where it's needed

Reply:
The newmessageform gets thrown away and recreated when you start a call, that is why I moved the initialization of the encoder up at "entry point" level. Even if we would use a service file, the `encoderReady` value would need to be stored globally.

Reply:
encoderService.js
```
import ...

let encoderReady = false
async function initialize() {
    if (encoderReady) return
    await register(connect())
}

export default initialize
```

Reply:
Actually that's probably what we should do, I just realised that my code has a bug. Audiorecordings won't work for the talk chat. I'll write the value in a store, sounds good?

Reply:
for a flag that only changes once from false to true and never again, I think it's ok to have it stored inside that service file

the benefit is greater as we only need to call initialize() in the one place we need it and not spread across the code with even extra events

Reply:
> Actually that's probably what we should do, I just realised that my code has a bug. Audiorecordings won't work for the talk chat. I'll write the value in a store, sounds good?

in that case transform the proposed "initialize" function into a store action, that would do.
the action could then be called multiple times but it would only initialize once